code/modules/mob/transform_procs.dm

```diff
@@ -156,27 +154,21 @@
 		O.equip_to_appropriate_slot(C)
 	qdel(animation)
 
-	O.gender = (deconstruct_block(getblock(dna.uni_identity, DNA_GENDER_BLOCK), 2)-1) ? FEMALE : MALE
-
-	if(dna)
-		dna.transfer_identity(O)
-		O.update_icons()
-		if(tr_flags & TR_KEEPSE)
-			O.dna.struc_enzymes = dna.struc_enzymes
-			domutcheck(O)
-
-	if(!dna.species)
-		hardset_dna(O, null, null, null, null, /datum/species/human)
-	else
-		hardset_dna(O, null, null, null, null, dna.species.type)
+	dna.transfer_identity(O)
+	O.updateappearance(mutcolor_update=1)
 
-	if(newname) //if there's a name as an argument, always take that one over the current name
-		O.real_name = newname
+	if(cmptext("monkey",copytext(O.dna.real_name,1,7)))
+		O.real_name = random_unique_name(O.gender)
+		O.dna.generate_unique_enzymes(O)
 	else
-		if(cmptext("monkey",copytext(O.dna.real_name,1,7)))
-			O.dna.real_name = random_unique_name(O.gender)
 		O.real_name = O.dna.real_name
-		O.name = O.real_name
+	O.name = O.real_name
+
+	if(tr_flags & TR_KEEPSE)
+		O.dna.struc_enzymes = dna.struc_enzymes
+		var/datum/mutation/human/race/R = mutations_list[RACEMUT] //we don't want to keep the race block active
+		O.dna.struc_enzymes = setblock(O.dna.struc_enzymes, R.dna_block, random_string(DNA_BLOCK_SIZE, list("0","1","2","3","4","5","6")))

```

Comment:
This ugly line is necessary to avoid having a human with the SE race block above 800, which happens when humanize() is called w/o updating the monkey's SE race block beforehand (changeling's lesser form, admin buttons)